gui/gui-manager.cpp

```diff
@@ -379,8 +365,14 @@ void GuiManager::runLoop() {
 			}
 		}
 
-		// Delay for a moment
-		_system->delayMillis(10);
+		_theme->updateScreen();
+
+		// Delay until the allocated frame time is elapsed to match the target frame rate
+		uint32 frameEndTime = _system->getMillis(true);
```

Comment:
`actualFrameDuration = _system->getMillis(true) - frameStartTime;` instead of storing the end time and subtracting twice.

Reply:
done